packages/material-ui/src/Select/SelectInput.js

```diff
@@ -276,7 +276,7 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
       >
         {/* So the vertical align positioning algorithm kicks in. */}
         {/* eslint-disable-next-line react/no-danger */}
-        {display || <span dangerouslySetInnerHTML={{ __html: '&#8203;' }} />}
+        {display != null ? display : <span dangerouslySetInnerHTML={{ __html: '&#8203;' }} />}
```

Comment:
We really should make it a habit to use explicit equality checks. 99% of the time you don't actually check for 0 or '' but just for undefined or null. The code should show this intention.